### DISCUSSION TITLE
2.3.6 代码  batch_size 调整后 结果

### DISCUSSION THREAD

**Discussion by GeLee-Q** [mlc-ai/mlc-zh#20] [Show and tell]
![image](https://user-images.githubusercontent.com/22428217/176416077-9630a3c1-6397-4e15-8d1b-b924157e913b.png)
浅浅的上传一下实验结果，可能粒度不够细致

**Comment by zzq96** [comment]
为什么64的时候最快那?难道L1 cache只有0.25KB(64 x 4 = 256B=0.25KB)?

**Comment by GeLee-Q** [comment]
https://tvm.apache.org/docs/how_to/optimize_operators/opt_gemm.html#blocking
首先查看官方文档，知道block是如何计算大小的

> A important trick to enhance the cache hit rate is blocking — data chunk will be computed block by block. The memory access inside the block is a small neighbourhood which is with high memory locality. In this tutorial, I picked up 32 as the blocking factor. So the block will fill 32 * 32 * sizeof(float) which is 4KB in the cache whose total size is 32KB (L1 data cache)

设置为32 就是 32 * 32 * 4 = 4K
设置为64 就是 64 * 64 * 4 = 16K
设置为 128 就 就是64K 超出了L1 Cache 32K的大小，所以64是最佳

**Reply by SilkSong** [reply]
我在本地跑了一下，发现每次结果都不太一样，best block size 有时是256，有时是1024，并不是64，是否是因为电脑不同导致的？
<img width="312" alt="Screen Shot 2022-06-29 at 23 54 51" src="https://user-images.githubusercontent.com/12102526/176559185-38cc9a5a-902a-40be-bd8f-4580b006062d.png">
